packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -615,13 +625,18 @@ class AndroidDevice extends Device {
 
   static final RegExp _timeRegExp = RegExp(r'^\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d{3}', multiLine: true);
 
-  /// Return the most recent timestamp in the Android log or null if there is
+  /// Return the most recent timestamp in the Android log or [null] if there is
   /// no available timestamp. The format can be passed to logcat's -T option.
   String get lastLogcatTimestamp {
-    final String output = runAdbCheckedSync(<String>[
-      'shell', '-x', 'logcat', '-v', 'time', '-t', '1',
-    ]);
-
+    String output;
+    try {
+      output = runAdbCheckedSync(<String>[
+        'shell', '-x', 'logcat', '-v', 'time', '-t', '1',
+      ]);
+    } catch (error) {
+      printError('Unexpected failure from adb: $error.');
+      return null;
```

Comment:
Previously this would exit, does the calling code handle null? Should we be doing a ToolExit here (and elsewhere?)

Reply:
The documentation indicates that null is expected when there is no timestamp.

Reply:
The calling code does handle null.